Baymax/Logger/Logger.swift

```
@@ -128,4 +127,62 @@ public struct Logger: TextOutputStream {
             }
         }
     }
+
+    /// Fetch log files on the `logQueue` and complete on the `.main` queue
+    /// 
+    /// - Parameter completion: Completion handler
+    func logFiles(completion: @escaping (Result<[LogFile], Error>) -> Void) {
+        guard let directory = directory else {
+            completion(.failure(LogError.directoryNotFound))
+            return
+        }
+
+        logQueue.async {
+            let result = Result({ try logFiles(in: directory) })
+            DispatchQueue.main.async {
+                completion(result)
+            }
+        }
+    }
+
+    /// Fetch `LogFile` in `directory`
+    ///
+    /// - Parameter directory: `URL`
+    private func logFiles(in directory: URL) throws -> [LogFile] {
+        let contents = try fm.contentsOfDirectory(
+            at: directory,
+            includingPropertiesForKeys: [.fileSizeKey, .creationDateKey],
+            options: []
+        )
+
+        let logFiles = try contents.compactMap { url -> LogFile in
+            let resourceValues = try url.resourceValues(
```

I have made a change here where this now throws on rather than fails silently